lib/core/ringbuffer.cpp

```diff
@@ -2,8 +2,33 @@
 
 typedef std::lock_guard<std::recursive_mutex> buffer_lock;
 
-RingBuffer::RingBuffer(size_t sz, metadataPool* pool, const std::string& _buffer_name,
-                       const std::string& _buffer_type) :
+static void print_py_status(RingBuffer* rb) {
```

Comment:
I would use `std::ostringstream` here instead of string concatenation.

Reply:
Good idea, done!